src/charm.py

```diff
@@ -80,11 +85,45 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _patch_auth_resources(self):
+        """Returns True if existing auth resources for training-operator are patched correctly."""
+        self.unit.status = MaintenanceStatus("Patching auth configuration")
+        client = Client()
+        context = {"namespace": self._namespace}
+
+        with open(os.path.join(self._src_dir, "auth_manifests.yaml")) as f:
+            for obj in codecs.load_all_yaml(f, context=context):
+                try:
+                    client.patch(
+                        type(obj), obj.metadata.name, obj, patch_type=PatchType.MERGE
+                    )
+                except ApiError as e:
+                    logging.error(f"Patching auth resources failed: {str(e)}")
+        logging.info("Auth resources successfully patched.")
+        return True
+
+    def _create_crds(self):
+        """Returns True if CRDs for training-operator are created correctly."""
+        self.unit.status = MaintenanceStatus("Creating CRDs")
+        client = Client()
+
+        with open(os.path.join(self._src_dir, "crds_manifests.yaml")) as f:
+            for obj in codecs.load_all_yaml(f):
+                try:
+                    client.create(obj)
+                except ApiError as e:
+                    logging.error(f"Creating CRDs failed: {str(e)}")
+        logging.info("CRDs successfully created.")
+        return True
```

Comment:
As above, this function always returns `True`...

Reply:
Same as above.